### PR TITLE
endpoint config resource unit tests

### PR DESCRIPTION
**Pull request by surajkota**
Description of changes:
- unit tests for endpoint config
- sdk.go is at 90%, overall endpoint config pkg is at 68.5%

Testing:
`make test`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/endpoint_config/testdata/test_suite.yaml [hunk 1]
```diff
@@ -1,17 +1,119 @@
-tests:
+  tests:
```
Comment (ryansteakley) [inline pkg/resource/endpoint_config/testdata/test_suite.yaml:1]: extra whitespace?

File: pkg/resource/endpoint_config/testdata/test_suite.yaml [hunk 2]
```diff
@@ -1,17 +1,119 @@
...
+        given:
+          desired_state: "v1alpha1/readone/desired/right_after_create.yaml"
+          svc_api:
+            - operation: DescribeEndpointConfigWithContext
+              output_fixture: "sdkapi/describe/describe_success.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/desired/right_after_create.yaml"
+      - name: "ReadOne=SuccessClearsConditions"
+        description: "Testing successful api call after an error right clears the conditions."
```
Comment (ryansteakley) [inline pkg/resource/endpoint_config/testdata/test_suite.yaml:74]: nit: Unclear description

Comment (surajkota) [reply pkg/resource/endpoint_config/testdata/test_suite.yaml]: yes reworded, it tests that a successful reconciliation clears conditions if  terminal/recoverable condition is already set

**Review by ryansteakley — APPROVED** [review]

**Pull request merged**